backend/model_manager.py

```
@@ -706,16 +722,28 @@ def delete_model(model_path):
-            return False
+    # Resolve against the directory listing so the path handed to os.remove
+    # always originates from MODELS_DIR enumeration, not from the request.
+    # String-only normalisation: no filesystem call touches the raw input.
+    try:
+        candidate = os.path.normcase(os.path.abspath(os.path.normpath(model_path)))
+    except (ValueError, TypeError):
+        return False
+    try:
+        entries = os.listdir(MODELS_DIR)
+    except OSError as e:
+        _logger.error("Error listing models directory: %s", e)
```

Ruff (0.15.20)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)